cli/lsp/performance.rs

```diff
@@ -129,14 +133,38 @@ impl Performance {
       .collect()
   }
 
+  pub fn do_logging(&mut self, enable: bool) {
+    self.logging_enabled = enable;
+  }
+
   /// Marks the start of a measurement which returns a performance mark
   /// structure, which is then passed to `.measure()` to finalize the duration
   /// and add it to the internal buffer.
-  pub fn mark<S: AsRef<str>>(&self, name: S) -> PerformanceMark {
+  pub fn mark<S: AsRef<str>, V: Serialize>(
+    &self,
+    name: S,
+    maybe_args: Option<V>,
+  ) -> PerformanceMark {
     let name = name.as_ref();
     let mut counts = self.counts.lock().unwrap();
     let count = counts.entry(name.to_string()).or_insert(0);
     *count += 1;
+    if self.logging_enabled {
```

Comment:
Ideally this should use the conditional logging provided by the rust log / env_logger crate.